MapboxNavigation/RouteMapViewController.swift

```diff
@@ -395,124 +320,148 @@ class RouteMapViewController: UIViewController {
+                    if let line = feature as? MLNPolylineFeature {
+                        let roadNameRecord = self.roadFeature(for: line)
+                        currentShieldName = roadNameRecord.shieldName
+                        currentName = roadNameRecord.roadName
+                    } else if let line = feature as? MLNMultiPolylineFeature {
+                        let roadNameRecord = self.roadFeature(for: line)
+                        currentShieldName = roadNameRecord.shieldName
+                        currentName = roadNameRecord.roadName
+                    }
+                }
+            }
         }
 
-        let noAnimation = {
-            animate()
-            complete(true)
+        let hasWayName = currentName != nil || currentShieldName != nil
+        if smallestLabelDistance < 5, hasWayName {
+            if let currentShieldName {
+                self.navigationView.wayNameView.attributedText = currentShieldName
+            } else if let currentName {
+                self.navigationView.wayNameView.text = currentName
+            }
+            self.navigationView.wayNameView.isHidden = false
+        } else {
+            self.navigationView.wayNameView.isHidden = true
         }
-
-        guard duration > 0.0 else { return noAnimation() }
-        UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: animate, completion: complete)
     }
+	
+    // MARK: End Of Route
+	
+    func transitionToEndNavigation(with duration: TimeInterval, completion: ((Bool) -> Void)? = nil) {
+        self.view.layoutIfNeeded() // flush layout queue
+        NSLayoutConstraint.deactivate(self.navigationView.bannerShowConstraints)
+        NSLayoutConstraint.activate(self.navigationView.bannerHideConstraints)
 
-    fileprivate func populateName(for waypoint: Waypoint, populated: @escaping (Waypoint) -> Void) {
-        guard waypoint.name == nil else { return populated(waypoint) }
-        CLGeocoder().reverseGeocodeLocation(waypoint.location) { places, error in
-            guard let place = places?.first, let placeName = place.name, error == nil else { return }
-            let named = Waypoint(coordinate: waypoint.coordinate, name: placeName)
-            return populated(named)
+        self.mapView.enableFrameByFrameCourseViewTracking(for: duration)
+        self.mapView.setNeedsUpdateConstraints()
+
+        let animate = {
+            self.view.layoutIfNeeded()
+            self.navigationView.floatingStackView.alpha = 0.0
         }
+
+        let noAnimation = { animate(); completion?(true) }
+
+        guard duration > 0.0 else { return noAnimation() }
+
+        self.navigationView.mapView.tracksUserCourse = false
+        UIView.animate(withDuration: duration, delay: 0.0, options: [.curveLinear], animations: animate, completion: completion)
     }
-}
 
-// MARK: - UIContentContainer
+    func hideEndOfRoute(duration: TimeInterval = 0.3, completion: ((Bool) -> Void)? = nil) {
```

Comment:
Did anything change here or was this simply moved?

Reply:
just moved